### DISCUSSION TITLE
What is a simple way to count instances of a value in a cell?

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1150] [Q&A] [answered]
For example I have
```
GO:0009056
GO:0009058
GO:0051301
GO:0006397
GO:0006397
GO:0006397
GO:0006464
GO:0006464
GO:0006464
GO:0006464
GO:0006464
GO:0006464
```

and want 

```
GO:0009056   1
GO:0009058   1
GO:0051301   1
GO:0006397    3      
GO:0006464    6
```

**Comment by sr320** [comment, marked as answer]
Awesome Q - how about...

`sort | uniq -c | awk '{print $2,$1}'`

**Comment by kubu4** [comment]
In a cell? Jupyter Notebook? If yes, can you share URL?

Can you share code that gives you that output?

**Reply by laurahspencer** [reply]
For R users - here's some code to convert a list of semicolon-separated GO terms per gene into separate rows for each GO term, then count how many GO terms there are per a designated "group") which you could use for this instance. Attached is the "Adult_FB_DEGs.csv" file to test the code. 
```
read.csv(file="Adult_FB_DEGs.csv") %>% 
  mutate(GO = strsplit(as.character(Gene.ontology.IDs), ";")) %>% 
    unnest(GO) %>% group_by(Entry) %>% count(GO) #remove "group_by" to just count total # of each GO terms 
```
[Adult_FB_DEGs.csv.zip](https://github.com/RobertsLab/resources/files/6152208/Adult_FB_DEGs.csv.zip)
